Assets/SEE/Utils/ReferenceEqualityComparer.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace SEE.Utils
+{
+    // NOTE: The below class was copied from the .NET 5.x source code.
```

Comment:
Are we allowed to do that?

Reply:
.NET is [licensed under MIT](https://github.com/dotnet/runtime/blob/main/LICENSE.TXT), so that should be fine, but we need to add the MIT license disclaimer here. I'll do so while working on the other comments.

Reply:
Thanks.